driver-core/src/main/java/com/datastax/driver/core/PoolingOptions.java

```diff
@@ -400,30 +406,48 @@ public PoolingOptions setIdleTimeoutSeconds(int idleTimeoutSeconds) {
     }
 
     /**
-     * Returns the timeout when trying to acquire a connection from a host's pool.
-     *
-     * @return the timeout.
+     * @deprecated see {@link #setPoolTimeoutMillis(int)}. This method always returns 0.
      */
+    @Deprecated
     public int getPoolTimeoutMillis() {
-        return poolTimeoutMillis;
+        return 0;
+    }
+
+    /**
+     * @deprecated the connection pool does not use a timeout anymore, incoming requests are now throttled with a
+     * threshold on the {@link #setMaxQueueSize(int)}  queue size}. This method has no effect.
+     */
+    @Deprecated
+    public PoolingOptions setPoolTimeoutMillis(int poolTimeoutMillis) {
+        return this;
     }
 
     /**
-     * Sets the timeout when trying to acquire a connection from a host's pool.
+     * Returns the maximum number of requests that get enqueued if no connection is available.
+     *
+     * @return the maximum queue size.
+     */
+    public int getMaxQueueSize() {
```

Comment:
Would it be worth giving this parameter a more meaningful name, e.g. `maxPendingConnectionRequests`, and making it clear in the javadocs that this number is meant _per connection pool_ (and not global)?